src/osd/OSD.cc

```diff
@@ -1436,12 +1436,13 @@ bool OSDService::prepare_to_stop()
 	osdmap->get_epoch(),
 	true  // request ack
 	));
-    utime_t now = ceph_clock_now();
-    utime_t timeout;
-    timeout.set_from_double(now + cct->_conf->osd_mon_shutdown_timeout);
-    while ((ceph_clock_now() < timeout) &&
+    mono_time start = mono_clock::now();
+    // make_timespan is in seconds
+    ceph::timespan dur = ceph::make_timespan(cct->_conf->osd_mon_shutdown_timeout);
+    mono_time timeout = start + dur;
+    while (mono_clock::now() < timeout &&
        (get_state() != STOPPING)) {
-      is_stopping_cond.WaitUntil(is_stopping_lock, timeout);
+      is_stopping_cond.WaitInterval(is_stopping_lock, dur);
```

Comment:
@mogeb i guess the reason why you switched over to `WaitInterval()` is that we don't have the overload for `WaitUntil(..., const std::chrono::time_point<>)`. but strictly speaking they are not equivalent in this context. can we go a step further? i mean, we can use `ceph::mutex` and `ceph::condition_variable` here. it's probably not in the scope of your change. but it helps us to achieve a more correct behavior.

what do you think?

Reply:
@tchaikov you're right, that is why I moved to `WaitInterval()`. From what I can tell from the code and my tests on a vstart cluster, the same behavior is achieved with both. Per your comment, I will look into refactoring this part using `ceph::condition_variable`

Reply:
@mogeb thank you!